test/infer/test_svi.py

```diff
@@ -232,3 +233,267 @@ def guide():
+
+    class FakeBeta(dist.Beta):
+        reparametrized_params = []
+
+    def model():
+        p_latent = numpyro.sample("p_latent", FakeBeta(alpha0, beta0))
+        with numpyro.plate("data", len(data)):
+            numpyro.sample("obs", dist.Bernoulli(p_latent), obs=data)
+        return p_latent
+
+    def guide():
+        alpha_q_log = numpyro.param("alpha_q_log", log_alpha_n + 0.17)
+        beta_q_log = numpyro.param("beta_q_log", log_beta_n - 0.143)
+        alpha_q, beta_q = jnp.exp(alpha_q_log), jnp.exp(beta_q_log)
+        p_latent = numpyro.sample("p_latent", FakeBeta(alpha_q, beta_q))
+        with numpyro.plate("data", len(data)):
+            pass
+        return p_latent
+
+    adam = optim.Adam(step_size=0.0007, b1=0.95, b2=0.999)
+    svi = SVI(model, guide, adam, loss=TraceGraph_ELBO())
+    svi_result = svi.run(jax.random.PRNGKey(0), 3000)
+
+    alpha_error = jnp.sum(
+        jnp.power(log_alpha_n - svi_result.params["alpha_q_log"], 2.0)
+    )
+    beta_error = jnp.sum(jnp.power(log_beta_n - svi_result.params["beta_q_log"], 2.0))
+
+    assert_allclose(alpha_error, 0, atol=0.03)
+    assert_allclose(beta_error, 0, atol=0.04)
+
+
+def test_tracegraph_gamma_exponential():
+    # exponential-gamma model
+    # gamma prior hyperparameter
+    alpha0 = 1.0
+    # gamma prior hyperparameter
+    beta0 = 1.0
+    n_data = 2
+    data = jnp.array([3.0, 2.0])  # two observations
+    alpha_n = alpha0 + n_data  # posterior alpha
+    beta_n = beta0 + data.sum()  # posterior beta
+    log_alpha_n = jnp.log(alpha_n)
+    log_beta_n = jnp.log(beta_n)
+
+    class FakeGamma(dist.Gamma):
+        reparametrized_params = []
+
+    def model():
+        lambda_latent = numpyro.sample("lambda_latent", FakeGamma(alpha0, beta0))
+        with numpyro.plate("data", len(data)):
+            numpyro.sample("obs", dist.Exponential(lambda_latent), obs=data)
+        return lambda_latent
+
+    def guide():
+        alpha_q_log = numpyro.param("alpha_q_log", log_alpha_n + 0.17)
+        beta_q_log = numpyro.param("beta_q_log", log_beta_n - 0.143)
+        alpha_q, beta_q = jnp.exp(alpha_q_log), jnp.exp(beta_q_log)
+        numpyro.sample("lambda_latent", FakeGamma(alpha_q, beta_q))
+        with numpyro.plate("data", len(data)):
+            pass
+
+    adam = optim.Adam(step_size=0.0007, b1=0.95, b2=0.999)
+    svi = SVI(model, guide, adam, loss=TraceGraph_ELBO())
+    svi_result = svi.run(jax.random.PRNGKey(0), 8000)
+
+    alpha_error = jnp.sum(
+        jnp.power(log_alpha_n - svi_result.params["alpha_q_log"], 2.0)
+    )
+    beta_error = jnp.sum(jnp.power(log_beta_n - svi_result.params["beta_q_log"], 2.0))
+
+    assert_allclose(alpha_error, 0, atol=0.04)
+    assert_allclose(beta_error, 0, atol=0.04)
+
+
+@pytest.mark.parametrize(
+    "num_latents,num_steps,step_size,atol,difficulty",
+    [
+        (3, 5000, 0.003, 0.05, 0.6),
+        (5, 6000, 0.003, 0.05, 0.6),
+        (7, 8000, 0.003, 0.05, 0.6),
```

Comment:
The numbers of steps are higher than the Pyro version because we don't use baseline here.